[
    {
        "content": "<p>Hello, </p>\n<p>My Zulip instance is stuck at version 1.4.3. Every upgrade I attempt breaks during the upgrade-sulip-stage-2, more specifically at: <br>\nApplying zerver.0041_create_attachments_for_old_messages </p>\n<p>For a while it was saying that a file was not found (an avatar - I already posted in this Production Help about it). </p>\n<p>Now I don't manage to understand the error message returned by the upgrade script. <br>\nTraceback (most recent call last):<br>\n  File \"/home/zulip/deployments/2018-07-11-14-17-39/scripts/lib/upgrade-zulip-stage-2\", line 115, in &lt;module&gt;<br>\n    subprocess.check_call([\"./manage.py\", \"migrate\", \"--noinput\"], preexec_fn=su_to_zulip)<br>\n  File \"/usr/lib/python2.7/subprocess.py\", line 540, in check_call<br>\n    raise CalledProcessError(retcode, cmd)<br>\nsubprocess.CalledProcessError: Command '['./manage.py', 'migrate', '--noinput']' returned non-zero exit status 1<br>\nTraceback (most recent call last):<br>\n  File \"/home/zulip/deployments/current/scripts/lib/upgrade-zulip\", line 53, in &lt;module&gt;<br>\n    subprocess.check_call([os.path.abspath(\"./scripts/lib/upgrade-zulip-stage-2\"), deploy_path])<br>\n  File \"/usr/lib/python2.7/subprocess.py\", line 540, in check_call<br>\n    raise CalledProcessError(retcode, cmd)<br>\nsubprocess.CalledProcessError: Command '['/home/zulip/deployments/2018-07-11-14-17-39/scripts/lib/upgrade-zulip-stage-2', '/home/zulip/deployments/2018-07-11-14-17-39']' returned non-zero exit status 1 </p>\n<p>Does anyone has any hint to offer?</p>",
        "id": 616939,
        "sender_full_name": "JPascalTaipei",
        "timestamp": 1532424514
    },
    {
        "content": "<p><span class=\"user-mention\" data-user-id=\"3310\">@JPascalTaipei</span> wow, that's an early version <span class=\"emoji emoji-1f603\" title=\"smiley\">:smiley:</span>.  Can you try first upgrading to 1.5.2, and then going from there?</p>",
        "id": 617080,
        "sender_full_name": "Tim Abbott",
        "timestamp": 1532447306
    },
    {
        "content": "<p>I'm pretty sure that for upgrading from 1.4.3, you have to upgrade to 1.5.2 first, and then upgrade further from there, because that migration (0041) imports code from the rest of the project that has changed since then.</p>",
        "id": 617082,
        "sender_full_name": "Tim Abbott",
        "timestamp": 1532447502
    },
    {
        "content": "<p>(Also, if that doesn't work, it'd be helpful to have the traceback)</p>",
        "id": 617083,
        "sender_full_name": "Tim Abbott",
        "timestamp": 1532447716
    },
    {
        "content": "<p>(Edited to suggest 1.5.2, since that has a few bug fixes for the 1.5.0 upgrade process)</p>",
        "id": 617084,
        "sender_full_name": "Tim Abbott",
        "timestamp": 1532447785
    }
]